File: Lib/Structure/Token.cs
namespace Miko.Lib.Structure
{
    public class Token
    {
        public Token(TokenType type, string Value, int Line, int Column)
        {
            this.Type = type;
            this.Value = Value;
            this.Line = Line;
            this.Column = Column;
        }

        public TokenType Type;
        public string Value;
        public int Line;
        public int Column;
    }
}

File: src/token.cpp
#include "token.hpp"

#include <string>

using namespace Miko;

Token::Token(TokenType type, std::string value, int line, int columnStart, int columnEnd)
{
    this->Type = type;
    this->Value = value;
    this->Line = line;
    this->ColumnStart = columnStart;
    this->ColumnEnd = columnEnd;
}